packages/mdc-menu/simple/foundation.js

```diff
@@ -399,69 +335,202 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
     }, numbers.SELECTED_TRIGGER_DELAY);
   }
 
-  /** @private */
-  autoPosition_() {
-    if (!this.adapter_.hasAnchor()) {
-      return;
-    }
+  /**
+   * @return {AutoLayoutMeasures} Measurements used to position menu popup.
+   */
+  getAutoLayoutMeasures_() {
+    const anchorRect = this.adapter_.getAnchorDimensions();
+    const viewport = this.adapter_.getWindowDimensions();
+
+    return {
+      anchorRect: anchorRect,
+      viewport: viewport,
+      viewportMargin: {
```

Comment:
Is this actually describing distance from anchor to edge of viewport? Maybe `viewportDistance` is a better term?

Reply:
keeping my name, unless you feel strongly about that.

Reply:
`Margin` tends to denote a gap of space where things _aren't_ supposed to appear. The reason I suggested `Distance` instead is because when I was first reading the code that used this, I was thinking this referred to some magical value of pixels we wanted to keep the menu away from the edge of the viewport; it took a while to dawn on me that it actually is strictly the distance to the edge of the viewport from the anchor element.

Reply:
Are you planning to rename this to `viewportDistance` from the earlier discussion?